modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/result/AbstractReturnDelegate.java

```diff
@@ -149,13 +148,12 @@ protected Message toMessage(Object value, ExecutionContextAdapter operationConte
       if (resultValue.getOutput() instanceof InputStream) {
         ConnectionHandler connectionHandler = (ConnectionHandler) operationContext.getVariable(CONNECTION_PARAM);
         if (connectionHandler != null && supportsStreaming(operationContext.getComponentModel())) {
-          resultValue = resultValue
-              .copy().output(componentDecoratorFactory
-                  .decorateOutput(new ConnectedInputStreamWrapper((InputStream) resultValue.getOutput(), connectionHandler,
-                                                                  getIncrementActiveComponent(operationContext),
-                                                                  getDecrementActiveComponent(operationContext)),
-                                  event.getCorrelationId()))
-              .build();
+          ConnectedInputStreamWrapper streamWrapper =
+              new ConnectedInputStreamWrapper((InputStream) resultValue.getOutput(), connectionHandler,
+                                              getDecrementActiveComponentTask(operationContext));
+          incrementActiveComponent(operationContext);
```

Comment:
incrementing this is not the return delegate's responsibility:

1) this class is used in other context to transform results to messages, and those shouldn't mess with the counters
2) If something fails AFTER the return delegate execution, nobody decrements it

The incrementing needs to happen in some other place, probably an interceptor which decrements it if the operation fails.